### PR TITLE
HDPath: add .parent() method.

### PR DESCRIPTION
**Pull request by msgilligan**
The motivation for this addition can be seen in this commit from a dependent draft PR: 15cccc34c820c41bc550b4343e76f2fff66805ec

### CODE REVIEW

File: core/src/test/java/org/bitcoinj/crypto/HDPathTest.java [hunk 1]
```diff
@@ -58,6 +58,22 @@ public void testExtendVarargs() {
...
 
+    @Test
+    public void testParent() {
+        HDPath path1 = HDPath.parsePath("m/0H/1H");
+
+        assertEquals(HDPath.parsePath("m/0H"), path1.parent());
+
+        HDPath path2 = HDPath.parsePath("m/0H");
+
+        assertEquals(HDPath.parsePath(""), path2.parent());
```
Comment (schildbach) [inline core/src/test/java/org/bitcoinj/crypto/HDPathTest.java:69]: I'm not too familiar with these paths, but should this parent not be 'm'?

File: core/src/test/java/org/bitcoinj/crypto/HDPathTest.java [hunk 2]
```diff
@@ -58,6 +58,22 @@ public void testExtendVarargs() {
...
+
+        HDPath path3 = HDPath.parsePath("");
+
+        assertEquals(HDPath.parsePath(""), path3.parent());
```
Comment (schildbach) [inline core/src/test/java/org/bitcoinj/crypto/HDPathTest.java:73]: We could also throw an exception in this case? What's the reason for making this edge case legal?

File: core/src/test/java/org/bitcoinj/crypto/HDPathTest.java (hunk 1 above)
Comment (msgilligan) [reply core/src/test/java/org/bitcoinj/crypto/HDPathTest.java:69]: It's actually an empty path at that point. The string representation is a little confusing. But thing of the underlying `List<ChildNumber>`. That is an empty list.

File: core/src/test/java/org/bitcoinj/crypto/HDPathTest.java (hunk 2 above)
Comment (msgilligan) [reply core/src/test/java/org/bitcoinj/crypto/HDPathTest.java:73]: I generally try to avoid exceptions and nulls, when possible. In this case, I think it make sense to return an empty list. In most use cases I expect the code will check before calling `.parent()`, but it is also possible to check for empty list after calling parent.

Comment (msgilligan) [reply core/src/test/java/org/bitcoinj/crypto/HDPathTest.java:73]: If look at https://github.com/bitcoinj/bitcoinj/commit/15cccc34c820c41bc550b4343e76f2fff66805ec (lines 904-905 in the changed file) you'll see the check for `!path.isEmpty()` before calling `.parent()`.

File: core/src/test/java/org/bitcoinj/crypto/HDPathTest.java (hunk 1 above)
Comment (schildbach) [reply core/src/test/java/org/bitcoinj/crypto/HDPathTest.java:69]: Ok, agreed.

File: core/src/test/java/org/bitcoinj/crypto/HDPathTest.java (hunk 2 above)
Comment (msgilligan) [reply core/src/test/java/org/bitcoinj/crypto/HDPathTest.java:73]: JavaDoc updated to explain this behavior more clearly.

**Comment by msgilligan** [comment]
@schildbach comments added, force-pushed, and ready to re-review/merge.

**Comment by msgilligan** [comment]
One more force push, I discovered a typo (extra '*') in a comment and removed it.

**Comment by schildbach** [comment]
Merged.
